skyportal/tests/utils/test_cache.py

```diff
@@ -0,0 +1,73 @@
+import shutil
+import os
+import time
+from os.path import join as pjoin
+
+import pytest
+
+from skyportal.utils.offset import Cache
+
+
+@pytest.fixture(scope="module")
+def cache_parent_dir(tmpdir_factory):
+    cache_parent = tmpdir_factory.mktemp("cache_test")
+    yield cache_parent
+    shutil.rmtree(str(cache_parent))
+
+
+@pytest.fixture()
+def cache(cache_parent_dir):
+    cache_path = pjoin(cache_parent_dir, 'cache')
+    cache = Cache(cache_path, max_items=3)
+    yield cache
+    shutil.rmtree(cache_path)
+
+
+def test_cache_hit(cache):
+    # Ensure cache directory was created
+    assert os.path.isdir(cache._cache_dir)
+
+    cache['some_key'] = b'abc'
+    fn = cache['some_key']
+    assert fn is not None
+
+
+def test_cache_max_items(cache):
+    cache['some_key'] = b'abc'
+    assert len(cache) == 1
+
+    cache['another_key'] = b'def'
+    assert len(cache) == 2
+
+    cache['yet_another_key'] = b'ghi'
+    assert len(cache) == 3
+
+    cache['the_last_straw'] = b'jkl'
+    assert len(cache) == 3
```

Comment:
Unclear to me why this passes. Isn't there now 4 items in the cache?

Reply:
The cache size is set to 3, so that's the upper limit

Reply:
Ok.